### PR TITLE
Add e2e tests for `kytos_stats`

### LINKED ISSUE #235: feature: Update e2e tests for a new napp (`kytos_stats`)
This is related to [EP035](https://github.com/kytos-ng/kytos/blob/master/docs/blueprints/EP035.rst)

`flow_stats` will evolve to `kytos_stats` for kytos statistics and e2e tests need to be updated:

- Update the routes in the e2e test suite for `flow_stats` to `kytos_stats`.
- Include an e2e test for a new endpoint `v1/table/stats` using OvS.

### PR DESCRIPTION
**Pull request by gretelliz**
Closes #235 

### Summary

Add e2e tests for `kytos_stats` endpoints including the new endpoint `v1/table/stats` to request tables.

### Local Tests

N/A

### End-to-end Tests

N/A

### CODE REVIEW

**Comment by viniarck** [comment]
Also, let's run the entire test suite with this branch for completeness before merging this PR.

**Comment by gretelliz** [comment]
@viniarck, Thank you!
Sure, I am going to add that test now.

**Comment by gretelliz** [comment]
@viniarck, In the new e2e test to check `activate_count` as you suggested, I set a time with time.sleep(60). The field is not increasing, how long should I wait to see an increase?

**Review by viniarck — APPROVED** [review]
Nicely done @gretelliz. LGTM. 

cc'ing @italovalcy in case he also wants to review before we merge this PR

**Comment by viniarck** [comment]
This needs to land with https://github.com/amlight/kytos-docker/pull/30 (future reminder for whoever is merging)

Also, @gretelliz did you have the chance to run the test suite?

**Comment by gretelliz** [comment]
Yes, @viniarck. See:

```
+ python3 -m pytest --timeout=180 tests/test_e2e_70_kytos_stats.py
============================= test session starts ==============================
platform linux -- Python 3.9.2, pytest-7.2.0, pluggy-1.0.0
rootdir: /tests
plugins: timeout-2.1.0, rerunfailures-10.2, anyio-3.6.2
timeout: 180.0s
timeout method: signal
timeout func_only: False
collected 7 items

tests/test_e2e_70_kytos_stats.py .......                                 [100%]

=============================== warnings summary ===============================
test_e2e_70_kytos_stats.py: 17 warnings
  /usr/lib/python3/dist-packages/mininet/node.py:1121: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    return ( StrictVersion( cls.OVSVersion ) <

test_e2e_70_kytos_stats.py: 17 warnings
  /usr/lib/python3/dist-packages/mininet/node.py:1122: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    StrictVersion( '1.10' ) )

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
------------------------------- start/stop times -------------------------------
================== 7 passed, 34 warnings in 435.14s (0:07:15) ==================
```

File: tests/test_e2e_70_kytos_stats.py [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+        api_url = KYTOS_STATS + '/table/stats?table=0'  
+        response = requests.get(api_url)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        assert len(data) == len(topo_switches)
+        for sw in topo_switches:
+            assert len(data[sw]) == 1
+            assert '0' in data[sw]
+
+        api_url = KYTOS_STATS + '/table/stats?table=0&table=1'
```
Comment (italovalcy) [inline tests/test_e2e_70_kytos_stats.py:90]: Gretel, sorry if I missed something. At this point, you didn't add flows into Table 1. Thus, my question is: how would the kytos_stats napp return data for table 1 if we don't have flows for table 1?

File: tests/test_e2e_70_kytos_stats.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+    def test_015_packet_count(self):
+        """Test packet_count""" 
+
+        api_url = KYTOS_STATS + '/flow/stats'  
+        response = requests.get(api_url)
+        assert response.status_code == 200, response.text
+        data_flow = response.json()
+        for sw in data_flow:
+            if len(data_flow[sw]):
```
Comment (italovalcy) [inline tests/test_e2e_70_kytos_stats.py:122]: Instead of trying to find a non-empty flow, maybe we could add such flow and query it later for stats.

**Review by italovalcy — COMMENTED** [review]
Gretel, everything looks good on the PR you sent! It is excellent to see the progress of the kytos_stats and now the end-to-end tests to validate its behavior.

I added a few comments through the code, but two things that could be also investigated are:

1) when testing for statistics, it would be nice if we collect kytos_stats information before, create some traffic between hosts, and collect again. To make sure that the stats are accurate. Maybe this can be done in another task later (please feel free to create separe issues for enhancing the test)

2) when doing asserts it is good to add some text for failure, to help troubleshooting later. For inscance, on line 38 you could have something like:

```
        assert '00:00:00:00:00:00:00:01' in data, str(data)
```

Overall the PR looks great!

File: tests/test_e2e_70_kytos_stats.py (hunk 2 above)
Comment (gretelliz) [reply tests/test_e2e_70_kytos_stats.py:122]: @italovalcy, it seems very good to me, what do you think about adding that test as part of the new test and leaving this test as it is?

Comment (gretelliz) [reply tests/test_e2e_70_kytos_stats.py:122]: @italovalcy, I created an issue to add all your suggestions and close this PR here.

**Pull request merged**